subsys/modbus/Kconfig

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2020 PHYTEC Messtechnik GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+config MODBUS_RTU
+	bool "Modbus RTU support"
+	depends on SERIAL && SERIAL_HAS_DRIVER
+
+if MODBUS_RTU
+
+config MODBUS_RTU_BUFFER_SIZE
+	int "Modbus buffer size"
+	default 256
+	range 64 256
+	help
+	  Modbus buffer size.
+
+choice
+	prompt "Supported node roles"
+	default MODBUS_RTU_ROLE_CLIENT_SERVER
+	help
+	  Specify the type of supported node roles.
+
+config MODBUS_RTU_ROLE_CLIENT
+	bool "Client support"
+
+config MODBUS_RTU_ROLE_SERVER
+	bool "Server support"
+
+config MODBUS_RTU_ROLE_CLIENT_SERVER
```

Comment:
do we really need a config option for both, why cant we just have users select both Client & server support?

Reply:
We really need a config option for both. The user can select: server role, client role, or server and client role.